### PR TITLE
Use Dune for build

### PR DESCRIPTION
**Pull request by pxeger**

### CODE REVIEW

**Comment by pxeger** [comment]
I named all the `Metasyb_XYZ` modules `XYZ_`, and kept the `Syb_XYZ` ones as `XYZ`, in keeping with the apparent convention from https://www.cl.cam.ac.uk/~jdy22/papers/staged-generic-programming.pdf. However, some of the `Metasyb` modules wouldn't collide with the `Syb`, so they could have the trailing underscores removed. Do you think I should do that?

File: lib/bindings_.ml [hunk 1]
```diff
@@ -99,11 +99,11 @@ let send_case_req : type b t t_. {P:PS} -> (t, t_) reifyt -> t code -> (t_ -> P.
       (reifyk unrolledk, false)
 
 
-(* if the continuations are both static and equal then we can replace
+(* if the continuations are both static_ and equal then we can replace
```
Comment (yallop) [inline lib/bindings_.ml]: I'm not ecstatic_ about this change; it looks like the result of overeager search-and-replace.

**Comment by yallop** [comment]
> However, some of the `Metasyb` modules wouldn't collide with the `Syb`, so they could have the trailing underscores removed. Do you think I should do that?

Yes, that sounds good to me.  The fewer trailing underscores, the better

File: lib/bindings_.ml (hunk 1 above)
Comment (pxeger) [reply lib/bindings_.ml]: Oops, you're right. I'll give a thorough scan through for any others, and fix them.

File: lib/bindings_.ml [hunk 2]
```diff
@@ -88,7 +88,7 @@ and ('t, 't_, 'b, 'bsta) case_record = {
 (* The single prompt for case-insertion *)
 let crp : case_req prompt = new_prompt ()
 
-let send_case_req : type b t t_. {P:PS} -> (t, t_) reifyt -> t code -> (t_ -> P.t) -> P.t * bool =
+let send_case_req : type _b t t_. {P:PS} -> (t, t_) reifyt -> t code -> (t_ -> P.t) -> P.t * bool =
```
Comment (yallop) [inline lib/bindings_.ml]:
```suggestion
let send_case_req : type t t_. {P:PS} -> (t, t_) reifyt -> t code -> (t_ -> P.t) -> P.t * bool =
```

File: metaocaml-syb.opam [hunk 3]
```diff
@@ -8,16 +8,17 @@ homepage: "https://github.com/yallop/metaocaml-syb"
...
+    ["dune" "runtest" "-p" name "-j" jobs] {with-test}
+]
 depends: [
-   "ocamlfind" {build}
+   "dune" {build}
    "higher"
    "delimcc"
    "genlet"
-   "ounit"
+   # "ounit"
```
Comment (yallop) [inline metaocaml-syb.opam]:
```suggestion
```

**Review by yallop — CHANGES_REQUESTED** [review]

**Comment by yallop** [comment]
Thank you!

**Pull request merged**